community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/CypherStatementWithOptions.scala

```diff
@@ -43,6 +43,8 @@ object CypherStatementWithOptions {
           case VersionOption(v) =>
             val newVersion = mergeOption(version, CypherVersion(v), "Can't specify multiple conflicting Cypher versions")
             recurse(tail, newVersion, planner, runtime, updateStrategy, executionMode)
+          case p: PlannerPreParserOption if p.name == "greedy" =>
```

Comment:
I prefer using `GreedyPlannerOption.name` here to a literal string. But do we really need to parse it still? Is this just to provide a nicer error message than `unknown planner option`?